src/stemmabench/algorithms/stemma.py

```diff
@@ -0,0 +1,174 @@
+import json
+from pathlib import Path
+from typing import Dict, List, Union
+from stemmabench.algorithms.manuscript import Manuscript
+from stemmabench.algorithms.manuscript_base import ManuscriptBase
+from stemmabench.algorithms.utils import Utils
+#from stemmabench.algorithms.stemma_algorithm import StemmaAlgo
+import os
+
+
+class Stemma:
+    """Class that represents a stemma tree."""
+
+    def __init__(self,
+                 folder_path: str = None,
+                 edge_file: str = None,
+                 generation_info: dict = None) -> None:
+        """A class to perform variant generation.
+        To instansite the class use one of the build methods.
+
+        Args:
+            folder_path(str, Optional): The path to the folder that contains the texts.
+            edge_file (str, Optional): An edge file from which the tree can be built. 
+            !!! The labels used in the edge file must be the same as the name used for the text .txt names !!!
+            generation_info (dict, Optional): A dictionnary containing information about the stemma's generation.
+
+        Generation_info content:
+            config: The path to the config file used to generate the tree.
+            has_ground_truth: Boolean indicating if the true tree is known. If false the path_to_original attribute will indicate the 
+            estimated original text.
+        """
+        self._set_folder_path(folder_path=folder_path)
+        self._edge_file: str = edge_file
+        self._generation_info: dict = generation_info
+        self._root: ManuscriptBase = None
+        self._text_lookup: dict[str, ManuscriptBase] = {}
+        self._fitted: bool = False
+
+    @property
+    def root(self):
+        return self._root
+
+    @property
+    def folder_path(self):
+        return self._folder_path
+    
+    @property
+    def edge_file(self):
+        return self._edge_file
+    
+    @property
+    def fitted(self):
+        return self._fitted
+    
+    @property
+    def text_lookup(self):
+        return self._text_lookup
+    
+    @property
+    def generation_info(self):
+        return self._generation_info
+    
+    def _set_folder_path(self, folder_path: str):
+        """Checks that the folder path is an existing directory and sets the folder_path attribute.
+        
+        Args:
+            folder_path (str, Requiered): The path to the folder containing all the texts.
+
+        Raises:
+            ValueError: If the specified folder_path is not an existing directory.
+        
+        """
+        if folder_path != None and not os.path.isdir(folder_path):
```

Comment:
tu peux raccourcir juste en évaluant comme ça:
```suggestion
        if folder_path and not os.path.isdir(folder_path):
```